day1/es3/es3.qml

```diff
@@ -0,0 +1,82 @@
+import QtQuick 2.12
+import QtQuick.Window 2.12
+
+Window {
+    width: 640
+    height: 800
+    visible: true
+    title: qsTr("Chapters exercise")
+
+    Column {
+        id: allChapters
+
+        readonly property int animationTime: 300
+
+        function changeChapter(clickedChapter){
+
+            var otherChapters = [] // all the chapters different from the clicked chapter
+            var chapterExpanded = undefined // here we set the chapter expanded ( if is present one )
+
+            for(var i = 0; i < allChapters.children.length; i++) {
+
+                if(allChapters.children[i].id === clickedChapter.id && allChapters.children[i] !== clickedChapter)
+                    otherChapters.push(allChapters.children[i]) // collect all the chapters different from the clicked
+
+                if(allChapters.children[i].isExpanded)
+                    chapterExpanded = allChapters.children[i]
+            }
+
+            if(clickedChapter.isExpanded === true)
+                clickedChapter.isExpanded = false // if the chapter clicked is expanded the simply set to close
+            else {
+
+                // set all the chapters not clicked to be closed
+                for(var j = 0; j < otherChapters.length; j++)
+                    otherChapters[j].isExpanded = false
+
+                // if nobody else chapter is expanded now then set immediately to expand the clicked chapter, else expande with a delay
+                if(chapterExpanded === undefined)
+                    clickedChapter.isExpanded = true
+                else {
+                    timerForDelayOpeningChapter.chapter = clickedChapter
+                    timerForDelayOpeningChapter.start()
+                }
+            }
+        }
+
+        anchors.centerIn: parent
+
+        Timer {
+            id: timerForDelayOpeningChapter
+
+            property var chapter: undefined
+
+            interval: allChapters.animationTime
+
+            onTriggered: {
+                chapter.isExpanded = true
+            }
+        }
+
+        Chapter {
+            title: "title 0"
+            content: "Quel ramo del lago di Como, che volge a mezzogiorno, tra due catene non interrotte di monti, tutto a seni e a golfi, a seconda"
+            onClickChapter: allChapters.changeChapter(clickedChapter)
+            animationTime: allChapters.animationTime
```

Comment:
Io solitamente sono del parere che se ogni volta che una componente viene istanziata finisce per avere delle property bindate sempre agli stessi valori, allora perde di senso usare le property: una property dovrebbe avere lo scopo di differenziare diverse istanze di componenti, ma se come in questo caso `animationTime` è sempre uguale, allora non sto davvero personalizzando il comportamento, e a questo punto è preferibile avere il valore integrato dentro la componente stessa.

Il mio gusto personale va nella direzione secondo cui `un minor numero di righe da leggere comporta una più semplice lettura, e di conseguenza una più rapida e immediata comprensione`.

Reply:
fatto, spostata animationTime dentro content